pkgs/build-support/trivial-builders/default.nix

```diff
@@ -878,7 +878,7 @@ rec {
              ) + "-patched"
     , patches   ? []
     , postPatch ? ""
-    }: stdenvNoCC.mkDerivation {
+    }: if patches == [ ] && postPatch == "" then src else stdenvNoCC.mkDerivation {
```

Comment:
This should probably receive a comment otherwise we might forget what the purpose of the condition was.